Index: vendor/magento/module-customer/Model/Options.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Model/Options.php b/Model/Options.php
--- a/Model/Options.php
+++ b/Model/Options.php (date 1630424668470)
@@ -97,15 +97,14 @@
         if (empty($options)) {
             return false;
         }
-
         $result = [];
-        $options = explode(';', $options);
+        $options = array_filter(explode(';', $options));
         foreach ($options as $value) {
-            $result[] = $this->escaper->escapeHtml(trim($value)) ?: ' ';
+            $value = $this->escaper->escapeHtml(trim($value));
+            $result[$value] = $value;
         }
-
         if ($isOptional && trim(current($options))) {
-            $result = array_merge([' '], $result);
+            $result = array_merge([' ' => ' '], $result);
         }

         return $result;
